src/document.js

```diff
@@ -0,0 +1,254 @@
+        {ocrData.map(({tag, text}) =>
+          <DocumentTextSection title={tag} text={text} />
+        )}
+      </div>
+    )
+  }
+
+}
+
+class DocumentEmailSection extends React.Component {
+
+  render() {
+    let doc = this.props.doc || {}
+    let data = doc.content
+    let files = doc.children || []
+
+    if(data.type == 'email') {
+      return (
+        <div>
+          <p className="bg-faded doc-section-title">Email</p>
+          <table className="table table-sm">
+            <tbody>
+              <tr><td>From</td> <td>{data.from}</td></tr>
+              <tr><td>To</td> <td>{data.to.join(', ')}</td></tr>
+              <tr><td>Date</td> <td>{data.date}</td></tr>
+              <tr><td>Subject</td> <td>{data.subject || '---'}</td></tr>
+            </tbody>
+          </table>
+        </div>
+      )
+    }
+    return null
+  }
+
+}
+
+class DocumentFilesSection extends React.Component {
+
+  render() {
+    let {data, baseUrl, title} = this.props
+    let files = data.map(({id, filename, content_type, size}, index) => {
+      return (
+        <tr key={index}>
+          <td>
+            {id
+              ? <a href={`${baseUrl}/${id}`}>{filename}</a>
+              : <span>{filename}</span>
+            }
+          </td>
+          <td className="text-muted">{content_type}</td>
+          <td className="text-muted">{size}</td>
+          <td>
+            {id
+              ? <a href={`${baseUrl}/${id}/raw/`}
+                  title="Original file">
+                <i className="fa fa-file-o"></i>
+              </a>
+              : <code>-- broken link --</code>}
+          </td>
+        </tr>
+      )
+    })
+
+    return files.length > 0 && (
+      <div>
+        <p className="bg-faded doc-section-title">{title}</p>
+        <table className="table table-sm">
+          <tbody>{files}</tbody>
+        </table>
+      </div>
+    )
+  }
+
+}
+
+class DocumentTextSection extends React.Component {
+
+  render() {
+    let text = this.props.text
+    if(!text) return null
+
+    let expanded = (this.state || {}).expanded
+    let title = this.props.title
+
+    if(text.length <= 700) {
+      expanded = true
+    }
+
+    return (
+      <div>
+        <div className="bg-faded doc-section-title">{title}</div>
+        <div className="content">
+          {expanded
+           ? <pre>{text}</pre>
+           : <pre className="content-wrap"
+                  onClick={() => {this.setState({expanded: true})}}>
+              {text}
+            </pre>
+          }
+        </div>
+      </div>
+    )
+  }
+
+}
+
+class DocumentLoading extends React.Component {
+
+  render() {
+    return (
+      <div className="loading">
+        <i className="fa fa-spinner loading-animate" aria-hidden="true"></i>
+        <p><small>Loading</small></p>
+      </div>
+    )
+  }
+
+}
+
+
+class DocumentHeaderLinks extends React.Component {
+
+  render() {
+    let data = this.props.data
+    if(data) {
```

Comment:
So this whole component only makes sense if `data` is not null. Why not invoke the component only if we have data? Also, you could make it a stateless component:

```javascript
function DocumentHeaderLinks({ data }) {
  return (
    <ul />
  )
}
```